libbeat/template/field.go

```diff
@@ -137,8 +137,9 @@ func (f *Field) array() common.MapStr {
 
 func (f *Field) object() common.MapStr {
 
+	dynProperties := f.getDefaultProperties()
+
 	if f.ObjectType == "text" {
```

Comment:
Each of these cases are mutually exclusive so I think a switch (or if/elseif) would be better.